### PR TITLE
Rid performTypeLocChecking of TypeLoc

### PR DESCRIPTION
**Pull request by AnthonyLatsis**

### CODE REVIEW

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

File: lib/SIL/Parser/ParseSIL.cpp [hunk 1]
```diff
@@ -1244,7 +1238,7 @@ bool SILParser::parseSILType(SILType &Result,
   
   // Apply attributes to the type.
   auto *attrRepr = P.applyAttributeToType(TyR.get(), attrs, specifier, specifierLoc);
-  auto Ty = performTypeLocChecking(attrRepr, /*IsSILType=*/true, OuterGenericEnv);
```
Comment (CodaFi) [inline lib/SIL/Parser/ParseSIL.cpp:1247]: This needs to use the full call in order to propagate `TypeResolutionFlags::SILType`.

**Review by CodaFi — APPROVED** [review]
Happy to see this go. Just one comment.

File: lib/SIL/Parser/ParseSIL.cpp (hunk 1 above)
Comment (AnthonyLatsis) [reply lib/SIL/Parser/ParseSIL.cpp:1247]: Oops

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
⛵

**Pull request merged**